### PR TITLE
Fix vizb rendering bugs, address round-2 issue and redesign sidebar

### LINKED ISSUE #450: gomlx_checkpoints --plot: same-metric-type lines collide, plus title/legend requests
## Context

Follow-up to #447 (merged). But there are few bugs,

## Bugs

1. **Same-metric-type lines don't show together correctly.** E.g. train loss and validation loss
   (same `MetricType="loss"`) should be two distinguishable lines on one chart, so you can see
   when they diverge — but they weren't rendering as expected.
2. **Multi-checkpoint comparison (2+ models) is broken.** One of the main use cases for `--plot`
   with multiple `<checkpoint_path>` args is comparing training runs side by side.

### Root cause (confirmed, not guessed)

Both bugs trace to the same cause. `vizb line --select step,colN{label}`'s `{label}` becomes the
`"type"` field in vizb's DataSet JSON — the field its chart renderer groups series by. That label
came from `plotLineInfo.short`, which wasn't guaranteed unique within one metric type.

In the real fixture, "Mean Loss on train" and "Mean Loss on validation" both produced the
identical `Short="#loss()"` (their eval dataset's `ShortName()` came out empty), so vizb silently
merged two different curves into one — confirmed by inspecting the intermediate DataSet JSON
(duplicate `"type"` keys within a single xAxis entry's `stats` array) and by an isolated repro
directly against `vizb line` with two `--select` columns sharing one `{label}`.

## Feature requests

3. `-plot_title` flag to title the generated page (handy for sharing results).
4. A legend box listing the models being compared (`#1`, `#2`, ...) and their directories.
5. A box mapping each metric's short label (e.g. `T/~loss`) to its full description (e.g. "Train:
   Moving Average Loss").

## Status

Fixed and implemented on branch `dev/vizb-plot-fixes` (not yet opened as a PR):

- `disambiguateShortLabels` detects Short collisions within a metric type and disambiguates only
  where needed — resolves both bugs.
- `-plot_title` flag.
- A fixed, independently-scrolling sidebar (not stacked above the charts) listing models compared
  and the short-label → long-name legend, with a color-coded dot per model (Okabe-Ito palette).
- 18 new tests; verified end-to-end against the maintainer's real fixture data.

PR to follow.

### PR DESCRIPTION
**Pull request by mayura-andrew**
Follow-up to #447, closes #450.

Real feedback from testing against actual FlowMatching training data (model-a/model-b).

## Rendering fix
Switched `runVizbLine` from vizb's "solo" mode to "group" mode (`-g step -p x --col-axis y`), so every line for a metric type overlays on one chart with a shared, clickable legend instead of vizb rendering one chart per series. Added `--smooth` for curved segments.

- Dropped "for model X" repetition in descriptions.
- Deduped the "Metric labels" glossary across models by `(coreShort, desc)`. Found and fixed a real determinism bug along the way: `createPlotLines` builds its line list from a Go map with randomized iteration order, so two models' disambiguation suffixes could get mismatched purely by chance, silently breaking the dedup. Fixed by sorting lines by `desc` first.
- Unicode model markers (❶/❷).

- Investigated X-axis categorical-vs-continuous and Y-axis log-scale reliability; both are genuine vizb limitations, reported upstream ([goptics/vizb#345](https://github.com/goptics/vizb/issues/345) for the Y-axis bug). Neither blocks this PR per the maintainer's own read: this can merge as-is now, with follow-up once vizb ships fixes.

## Value-rounding bug
Found `vizb line` was rounding every value to a fixed 2 decimal places regardless of magnitude, worked around temporarily, reported upstream, fixed by vizb in v0.18.0. Workaround removed once reconfirmed against the new version — `runVizbLine` now requires vizb >= v0.18.0.

## Sidebar redesign
- Restructured `legendEntry` into term/detail pairs rendered as `<dl>`/`<dt>`/`<dd>`.
- Validated colorblind-safe palette, dark-mode-aware tokens.
- "Metric labels" grouped into one section per metric type.
- Chart line colors now match sidebar swatches via vizb's `--theme` flag — verified end-to-end against the real fixture, zero mismatches across 19 series.
- Removed the swatch from "Models compared" once color started meaning "which metric" everywhere else, to avoid a model's dot coincidentally matching an unrelated metric's color.

## Test plan
- [x] 29 new/changed tests, all passing (`go test ./cmd/gomlx_checkpoints/...`)
- [x] `go vet` clean
- [x] Verified against the real model-a/model-b fixture
- [x] Verified against a freshly-generated single-model, 3-metric-type synthetic dataset

### CODE REVIEW

File: cmd/gomlx_checkpoints/plots.go [hunk 1]
```diff
@@ -103,7 +123,6 @@ func createPlotLines(metricType string, modelNames []string, metricsOrder map[Mo
 		modelName := modelNames[modelIdx]
 		modelNum := modelNamesToIndex[modelName]
 
-		// Group points by metric name.
```
Comment (janpfeifer) [inline cmd/gomlx_checkpoints/plots.go:106]: Nit: the AI seems to have removed these comments. This is unfortunate, since they are useful for someone reading the code.

Would it be possible to revert the various comments removal ?

**Review by janpfeifer — CHANGES_REQUESTED** [review]
Thanks @mayura-andrew , very appreciated.

One nit detail: would you mind preserving the comments in the `plots.go` ?

File: cmd/gomlx_checkpoints/plots.go (hunk 1 above)
Comment (mayura-andrew) [reply cmd/gomlx_checkpoints/plots.go:106]: Okay thank you @janpfeifer I will make these changes :)

Comment (mayura-andrew) [reply cmd/gomlx_checkpoints/plots.go:106]: Oops I did a mistake I will fix it.

Comment (mayura-andrew) [reply cmd/gomlx_checkpoints/plots.go:106]: I have committed all the unnecessary files, so I need some time for this.
